### ISSUE TITLE
Bind group and layout entries should be optional

### ISSUE THREAD

**Issue by nical** [gfx-rs/wgpu#4738]
Spec:

- https://gpuweb.github.io/gpuweb/#gpubindingcommandsmixin-setbindgroup
- https://www.w3.org/TR/webgpu/#dom-gpupipelinelayoutdescriptor-bindgrouplayouts

Bind group layouts and bind groups should be `null`able, which in Rust parlance means we should take an `Option` of them. This lets app authors “unset” a bind group at an index.

**Comment by jimblandy** [comment]
It's unclear what the underlying platform APIs' expectations are around removing bindgroups, so we're going to need to investigate this a bit before we can tell how much work it will be.

**Comment by teoxoy** [comment]
Relevant spec PR: https://github.com/gpuweb/gpuweb/pull/3938

**Comment by Wumpf** [comment]
this got fixed in
* https://github.com/gfx-rs/wgpu/pull/6216

**Comment by Wumpf** [comment]
ah I spoke too soon, HAL doesn't handle this yet it seems?

**Comment by ErichDonGubler** [comment]
Note that this should include optional [`PipelineLayoutDescriptor::bind_group_layouts`](https://docs.rs/wgpu/27.0.1/wgpu/struct.PipelineLayoutDescriptor.html#structfield.bind_group_layouts).

ETA: Folded into OP.

**Comment by ErichDonGubler** [comment]
Firefox downstream tracking: [bug 1992771](https://bugzilla.mozilla.org/show_bug.cgi?id=1992771)

**Comment by andyleiserson** [comment]
Also discussed in #1489. (I am not going to mark a duplicate right now because there is some discussion in both issues that might be worth consolidating.)
